File: Behavioural-Patterns/Mediator/C#/Mediator/Implementations/CacheService.cs
public class CacheService : Subscriber{

    public void onEvent(Object ev) {
        if (ev is UserLoggedInEvent) {
            UserLoggedInEvent loginEvent = (UserLoggedInEvent) ev;
           Console.WriteLine("   [CacheService] Invalidating cache for user: " + loginEvent.username);
        }
    }

    public String getName() {
        return "CacheService";
    }
    
}

File: Behavioural-Patterns/Mediator/Java/Implementations/CacheService.java
package Implementations;

import Interfaces.Subscriber;

public class CacheService implements Subscriber{

    @Override
    public void onEvent(Object event) {
        if (event instanceof UserLoggedInEvent) {
            UserLoggedInEvent loginEvent = (UserLoggedInEvent) event;
            System.out.println("   [CacheService] Invalidating cache for user: " + loginEvent.username);
        }
    }

    @Override
    public String getName() {
        return "CacheService";
    }
    
}
